code/modules/clothing/spacesuits/rig/rig.dm

```diff
@@ -302,8 +310,96 @@
 			piece.item_flags |=  (STOPPRESSUREDAMAGE|AIRTIGHT)
 	update_icon(1)
 
-/obj/item/weapon/rig/process()
+/obj/item/weapon/rig/ui_action_click()
+	toggle_cooling(usr)
+
+/obj/item/weapon/rig/proc/toggle_cooling(var/mob/user)
+	if(cooling_on)
+		turn_cooling_off()
+	else
+		turn_cooling_on()
+
+/obj/item/weapon/rig/proc/turn_cooling_on()
+	if(!cell)
+		return
+	if(cell.charge <= 0)
+		usr << "<span class='notice'>\The [src] has no power!.</span>"

```

Comment:
You can eliminate usage of usr by adding a user argument to both this and ```turn_cooling_off```.  ```toggle_cooling``` already has an argument for user that you can just feed into those two procs.